app/code/Magento/ConfigurableProduct/Pricing/Render/FinalPriceBox.php

```diff
@@ -68,7 +72,7 @@ public function __construct(
     public function hasSpecialPrice()
     {
         $product = $this->getSaleableItem();
-        foreach ($this->lowestPriceOptionsProvider->getProducts($product) as $subProduct) {
+        foreach ($product->getTypeInstance()->getUsedProducts($product) as $subProduct) {
```

Comment:
lowestPriceOptionsProvider was introduced for improve performance to not iterate over all product (it can be 5k for example)

Reply:
Thank you for your review!
 You've closed the issue but pull request still open, it means I can change the solution of this problem (for example for faster getting the result we can use right direct SQL query likes Magento\ConfigurableProduct\Pricing\Price\LowestPriceOptionsProvider::getProducts()) or I should give up this issue?